### DISCUSSION TITLE
Diffing headless browser and real browser?

### DISCUSSION THREAD

**Discussion by yourfavorite** [ulixee/secret-agent#238] [Q&A]
As part of this project (BrowserEmulators) I'm assuming there are tools used to help spot differences between the real browser and the headless browser. Would it be possible to get access to this type of tool to be used for other projects?

**Comment by blakebyrnes** [comment]
Everything you need is in Double Agent: https://github.com/ulixee/double-agent

The only portion we don't include is the part that runs this on machines to get the actual values out, but you could replicate on your own.

I think @calebjclark is going to open source the ScraperReport again once we've updated it, so that should also give you insight into how to run it for various engines.
